src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -556,10 +554,9 @@ def _report_single_requirement_conflict(self, req, parent):
 
     def get_installation_error(
         self,
-        e,  # type: ResolutionImpossible[Requirement, Candidate]
-        constraints,  # type: Dict[str, Constraint]
-    ):
-        # type: (...) -> InstallationError
+        e: ResolutionImpossible[Requirement, Candidate],
```

Comment:
This line throws an exception:

e: ResolutionImpossible[Requirement, Candidate],

Reply:
Because `ResolutionImpossible` is a conditional import and has different types during mypy check and runtime. Wrap it on quotes.

Reply:
@uranusjr Thanks for the comment, done!